src/Refactor/EntityWriter/AssignmentXmlWriter.php

```diff
@@ -0,0 +1,26 @@
+<?php
+namespace DemandwareXml\Refactor\EntityWriter;
+
+use DemandwareXml\Refactor\Entity\Assignment;
+use DemandwareXml\Refactor\Xml\{XmlFormatter, XmlWriter};
+
+class AssignmentXmlWriter
+{
+    private $writer;
+    private $assignment;
+
+    public function __construct(XmlWriter $writer, Assignment $assignment)
+    {
+        $this->writer     = $writer;
+        $this->assignment = $assignment;
+    }
+
+    public function write(): void
+    {
+        $this->writer->startElement('category-assignment');
+        $this->writer->writeAttribute('product-id', $this->assignment->productId);
+        $this->writer->writeAttribute('category-id', $this->assignment->categoryId);
+        $this->writer->ifNotEmpty()->writeElement('primary-flag', XmlFormatter::fromBoolean($this->assignment->primary));
```

Comment:
Why do we need `ifNotEmpty()`? Don't we already know it's not empty because of `startElement()` above?

Reply:
If you look at the `assignment.xml` fixture you'll see there are situations where we don't send this particular element: `<category-assignment product-id="PROD1" category-id="CAT42"/>`